### PR TITLE
fix(safe-claiming-app): validate custom delegation address

### LINKED ISSUE #540: [$SAFE Claiming App] Self-delegation
## Overview

The delegate registry does not allow self-delegation but we do not validate the input field and provide no information leading to failing transactions. 

## Requirements

Validate that provided address does not equal your own Safe address 
Show error message "You cannot delegate to your own Safe address" 
Change text above input field (design is already updated)

## Designs

https://www.figma.com/file/igarLtnMhVtrYNOATmjJDc/Claim-S-Tokens-App?node-id=1%3A2

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves
Resolves #540 

## How this PR fixes it

- Updates the input field text description
- Validates that the address is not the same as the current safe address

## How to test it

1. Open a safe that is elligible for the airdrop
2. Navigate to the custom delegate screen
3. Input the current safe address
4. Observe a validation error and the next button is disabled
5. Input an ENS that resolves to the current safe address
6. Observe a validation error and the next button is disabled 

## Screenshots
Screenshot 2022-10-04 at 11 31 55

### CODE REVIEW

File: apps/safe-claiming-app/src/hooks/useEnsResolution.ts [hunk 1]
```diff
@@ -57,6 +61,11 @@ export const useEnsResolution = (
         const resolvedName = await web3Provider.resolveName(customAddress)
 
         if (resolvedName !== null && ethers.utils.isAddress(resolvedName)) {
+          if (resolvedName === safe.safeAddress) {
```
Comment (DiogoSoaress) [inline apps/safe-claiming-app/src/hooks/useEnsResolution.ts:64]: `sameAddress` is safer perhaps

**Review by DiogoSoaress — CHANGES_REQUESTED** [review]

**Review by DiogoSoaress — APPROVED** [review]

**Review by schmanu — CHANGES_REQUESTED** [review]
The changes look 👀  good.

But there is a unit test for the `useEnsResolution` hook. 
You should add the 2 new created cases there: 
- address is safe address
- ens name resolves to safe address

**Review by schmanu — APPROVED** [review]
Perfect! Nice fix!

**Comment by liliya-soroka** [comment]
Verified

**Pull request merged**